Cython/Utility/ExtensionTypes.c

```diff
@@ -69,21 +64,43 @@ static int __Pyx_fix_up_extension_type_from_spec(PyType_Spec *spec, PyTypeObject
                     descr = PyDescr_NewMember(type, memb);
                     if (unlikely(!descr))
                         return -1;
-                    if (unlikely(PyDict_SetItem(type->tp_dict, PyDescr_NAME(descr), descr) < 0)) {
-                        Py_DECREF(descr);
+                    int set_item_result = PyDict_SetItem(type->tp_dict, PyDescr_NAME(descr), descr);
+                    Py_DECREF(descr);
+                    if (unlikely(set_item_result < 0)) {
                         return -1;
                     }
-                    Py_DECREF(descr);
                     changed = 1;
                 }
-#endif
             }
             memb++;
         }
         if (changed)
             PyType_Modified(type);
```

Comment:
We probably don't need to do this twice but can merge the two `changed` variables to do it at the end.